File: 26-spring-boot-listener/src/main/java/com/gzz/user/UserController.java
package com.gzz.user;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
/**
 * @author https://www.jianshu.com/u/3bd57d5f1074
 * @date 2019-12-24 10:50:00
 **/
@RestController
public class UserController {
	@Autowired
	private UserService userService;

	@RequestMapping("/register")
	public String register(User user) {
		userService.register(user);
		System.out.println("注册成功.");
		return "注册成功.";
	}
}

File: 99-spring-boot-kotlin/src/main/java/com/gzz/mvc/UserController.kt
package com.gzz.mvc

import org.springframework.web.bind.annotation.RestController
import org.springframework.beans.factory.annotation.Autowired
import javax.servlet.http.HttpServletRequest
import org.springframework.web.bind.annotation.RequestMapping
import com.gzz.mvc.User
import com.gzz.base.Page


@RestController
class UserController {

	@Autowired
	lateinit var service: UserService

	@RequestMapping("hello")
	fun page(): String {
		return "hello"
	}

	@RequestMapping("query")
	fun query(cond: UserCond): List<User> {
		return service.queryList(cond);
	}

	@RequestMapping("page")
	fun page(cond: UserCond): Page<User> {
		return service.queryPage(cond);
	}

}
